builder/steps/gen-dockerfile/contents/test/generate_files_test.ts

```diff
@@ -53,6 +54,12 @@ const INSTALL_NPM =
 RUN yarn global add npm@'${NPM_VERSION}'
 `;
 
+const INSTALL_YARN =
+    `# Install the version of yarn as requested by the engines.yarn field in
+# package.json.
+RUN yarn global add yarn@'${YARN_VERSION}'
```

Comment:
question: are the single quotes around ${YARN_VERSION} necessary?

Reply:
The single quotes are used to support semver strings with spaces (such as `>1.0 <2.0'`).